src/tools/export_file_tool.rb

```diff
@@ -58,29 +59,45 @@ def export_partslist
     number_hubs = Graph.instance.nodes.count
     number_bottles = {}
     number_actuators = {}
+    lengths_for_link = []
+    springs = []
     Graph.instance.edges.each_value do |edge|
-      if edge.link.is_a?(PhysicsLink)
+      if edge.link_type == 'actuator'
         # the actuator length is rounded to the closest 10 cm
         actuator_length = (edge.link.length/10.0).round * 10
         increase_number_of(number_actuators, actuator_length)
+      elsif edge.link_type == 'spring'
+        springs << edge.link
       else
         increase_number_of(number_bottles, edge.bottle_type)
       end
+      lengths_for_link.push(id: edge.inspect, length: (edge.length - 2 * Configuration::BALL_HUB_RADIUS).to_mm)
     end
 
     dir_name = File.dirname(@export_path)
     base_name = File.basename(@export_path, File.extname(@export_path))
+    puts "dir_name: #{dir_name}, base_name: #{base_name}"
```

Comment:
You wanna keep this in here? Or is it a debug relict...

Reply:
You never catch all of them :D